gokitmiddlewares/loggingmiddleware/middleware.go

```diff
@@ -100,3 +86,43 @@ func toString(i interface{}, n int) string {
 	}
 	return s[:n]
 }
+
+func doCustomEnrichRequest(
+	ctx context.Context,
+	config Config,
+	logger *zerolog.Logger,
+	request interface{},
+) context.Context {
+	if typedReq, ok := request.(LoggableEndpointRequest); ok {
+		logger.UpdateContext(func(zctx zerolog.Context) zerolog.Context {
+			ctx, zctx = typedReq.EnrichLog(ctx, zctx)
+			return zctx
+		})
+	} else if config.EnrichLogWithRequest != nil {
```

Comment:
I think we could remove the else here and allow to both enrich functions to be used. Otherwise you may get a strange behavior the you will set the enrich func but it will not execute.